relay-server/src/actors/store.rs

```diff
@@ -1049,3 +1103,162 @@ impl Message for KafkaMessage {
+            key_id: Some(17),
+        };
+
+        let attachment_vec = {
+            let item = Item::new(ItemType::Attachment);
+
+            vec![ChunkedAttachment {
+                id: Uuid::new_v4().to_string(),
+                name: UNNAMED_ATTACHMENT.to_owned(),
+                content_type: item
+                    .content_type()
+                    .map(|content_type| content_type.as_str().to_owned()),
+                attachment_type: item.attachment_type().cloned().unwrap_or_default(),
+                chunks: 0,
+                size: None,
+                rate_limited: Some(item.rate_limited()),
+            }]
+        };
+
+        (start_time, event_id, scoping, attachment_vec)
+    }
+
+    #[test]
+    fn test_return_attachments_when_missing_event_item() {
+        let (start_time, event_id, scoping, attachment_vec) = test_arguments_extract_kafka_msgs();
+
+        let mut kafka_messages = StoreService::extract_kafka_messages(
+            None,
+            event_id,
+            scoping,
+            start_time,
+            None,
+            attachment_vec,
+        );
+
+        assert!(kafka_messages.any(|msg| {
+            if let KafkaMessage::Attachment(_) = msg {
+                return true;
+            }
+            false
+        }));
+    }
+
+    /// If there is an event_item, and it is of type transaction, then it should strip the attachments
+    /// from the event and return them as standalone attachments (so it will be sent to kafka later).
+    #[test]
+    fn test_strip_attachment_from_transaction() {
+        let get_kafka_messages = || {
+            let (start_time, event_id, scoping, attachment_vec) =
+                test_arguments_extract_kafka_msgs();
+
+            let item = Item::new(ItemType::Transaction);
+            let event_item = Some(&item);
+
+            StoreService::extract_kafka_messages(
+                event_item,
+                event_id,
+                scoping,
+                start_time,
+                None,
+                attachment_vec,
+            )
+        };
+
+        let mut kafka_messages = get_kafka_messages();
+
+        // Checks that the attachments of the event is stripped
+        let event = kafka_messages
+            .find(|msg| {
+                if let KafkaMessage::Event(_) = msg {
+                    return true;
+                }
+                false
+            })
+            .unwrap();
+        if let KafkaMessage::Event(event) = event {
+            assert!(event.attachments.is_empty());
+        } else {
+            panic!("No event found")
+        }
+
+        let mut kafka_messages = get_kafka_messages();
+
+        // Checks that the attachment is a message on its own in the vector.
+        assert!(kafka_messages.any(|msg| {
+            if let KafkaMessage::Attachment(_) = msg {
+                return true;
+            }
+            false
+        }));
+    }
```

Comment:
nit: You could use [partition](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition) here to get both types of messages at once.